api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchController.java

```diff
@@ -1,21 +1,34 @@
 package com.google.search.googlesearch.controller.v1.api;
 
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
-
 import io.swagger.v3.oas.annotations.media.Content;
 import io.swagger.v3.oas.annotations.media.Schema;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class SearchController {
-    @GetMapping(value = "/api/v1/search")
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE, value = "v1/search")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "Get search results", content = @Content(schema = @Schema(nullable = true)))
     })
-    public String getSearch(@RequestParam(defaultValue = "param") String query) {
-        return "Query: " + query;
+    public @ResponseBody String searchResultv1(@RequestParam(defaultValue = "param") String query) {
+        String response = "";
+
+        if (query.equals("facebook")) {
+            response = "{\"total_results\":6,\"execution_time\":45000,\"query_string\":\"" + query + "\",\"results\":[{\"title\":\"Facebook - Log In or Sign Up\",\"url\":\"https://www.facebook.com/\",\"short_description\":\"Create an account or log into Facebook. Connect with friends, family and other people you know. Share photos and videos, send messages and get updates.\"},{\"title\":\"Facebook (@Facebook) · Twitter\",\"url\":\"https://twitter.com/Facebook\",\"short_description\":\"\"},{\"title\":\"Facebook – Apps on Google Play\",\"url\":\"https://play.google.com/store/apps/details?id=com.facebook.katana&hl=en_IN&gl=US\",\"short_description\":\"Keeping up with friends is faster and easier than ever. Share updates and photos, engage with friends and Pages, and stay connected to communities important ...\"},{\"title\":\"Newsroom | About Facebook\",\"url\":\"https://about.fb.com/news/\",\"short_description\":\"View the latest news, announcements and media resources from Facebook.\"},{\"title\":\"About Facebook\",\"url\":\"https://about.fb.com/\",\"short_description\":\"Facebook builds technologies that give people the power to connect with friends and family, find communities and grow businesses.\"},{\"title\":\"Facebook - Wikipedia\",\"url\":\"https://en.wikipedia.org/wiki/Facebook\",\"short_description\":\"Facebook (stylized as facebook) is an American online social media and social networking service based in Menlo Park, California, and a flagship service of the ...\"}]}";

```

Comment:
Follow layered architecture. 
`Controller -> Service -> Model -> Controller`

1. Keep the controller very simple. 
2. Move logics to the `Service` layer and have a `Model` for standardizing the return type. 
3. Use JSON reader instead of hardcoding 

Refer this PR - https://github.com/web-slate/simple-google-search/pull/18

Reply:
yes. This is plan to start with dirty approach